src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/ApplyBQSR.java

```diff
@@ -47,18 +47,23 @@
     
     private SAMFileGATKReadWriter outputWriter;
 
-    private ReadTransformer transform;
+    /**
+     * Returns the BQSR transformer
+     */
+    @Override
+    public ReadTransformer makeReadTransformer(){
+        return new BQSRReadTransformer(getHeaderForReads(), BQSR_RECAL_FILE, bqsrArgs);
+    }
 
     @Override
     public void onTraversalStart() {
         outputWriter = createSAMWriter(OUTPUT, true);
-        transform = new BQSRReadTransformer(getHeaderForReads(), BQSR_RECAL_FILE, bqsrArgs);
         Utils.warnOnNonIlluminaReadGroups(getHeaderForReads(), logger);
     }
 
     @Override
     public void apply( GATKRead read, ReferenceContext referenceContext, FeatureContext featureContext ) {
-        outputWriter.addRead(transform.apply(read));
+        outputWriter.addRead(read);
```

Comment:
I think that `ApplyBQSR` should continue to apply the read transformer manually, rather than via the new `GATKTool` interface, so that it can guarantee that it's applied only to filtered reads (since it could fail on malformed reads). 

There's a bit of an issue here with some read transformers that definitely should be applied before filtering (like `FixMisencodedBaseQualityReads`) to give an opportunity to repair bad input that would otherwise be filtered, and others (like `BQSRReadTransformer`) that should definitely be applied after filtering so that they don't fail on malformed input. We may eventually need a mechanism to specify whether a given transformer should be applied before or after filtering. Thoughts on this?